src/pathfinding_service/model/token_network.py

```diff
@@ -31,14 +33,20 @@ def __init__(
         self.nodes = nodes
         self.value = value
         self.address_to_reachability = address_to_reachability
+        self.fees: List[FeeAmount] = [
+            self.G[self.nodes[i]][self.nodes[i + 1]]["view"].fee_receiver(self.value)
+            + self.G[self.nodes[i + 1]][self.nodes[i + 2]]["view"].fee_sender(self.value)
+            for i, node in enumerate(self.nodes[1:-1])  # initiator and target don't cause fees
+        ]
 
     @property
     def edge_attrs(self) -> Iterable[dict]:
         return (self.G[node1][node2] for node1, node2 in zip(self.nodes[:-1], self.nodes[1:]))
 
     def to_dict(self) -> dict:
-        fee = sum(edge["view"].fee(self.value) for edge in self.edge_attrs)
-        return dict(path=[to_checksum_address(node) for node in self.nodes], estimated_fee=fee)
+        return dict(
+            path=[to_checksum_address(node) for node in self.nodes], estimated_fee=sum(self.fees)
```

Comment:
I think `sum` is wrong here, as the fees later in a given route have to be included in the earlier ones.